x/avs/keeper/impl_epoch_hook.go

```diff
@@ -22,12 +27,87 @@ func (k *Keeper) EpochsHooks() EpochsHooksWrapper {
 
 // AfterEpochEnd is called after an epoch ends. It is called during the BeginBlock function.
 func (wrapper EpochsHooksWrapper) AfterEpochEnd(
-	_ sdk.Context, _ string, _ int64,
+	ctx sdk.Context, epochIdentifier string, epochNumber int64,
 ) {
-	// get all the avs address bypass the epoch end
-	// _ = wrapper.keeper.GetEpochEndAVSs(ctx, epochIdentifier, epochNumber)
+	// get all the task info bypass the epoch end
+	// threshold calculation, signature verification, nosig quantity statistics
+	taskResList := wrapper.keeper.GetTaskStatisticalEpochEndAVSs(ctx, epochIdentifier, epochNumber)
+
+	if len(taskResList) != 0 {
+		groupedTasks := wrapper.keeper.GroupTasksByIDAndAddress(taskResList)
+		for _, value := range groupedTasks {
+			var signedOperatorList []string
+			var taskID uint64
+			var taskAddr string
+			var avsAddr string
+			operatorPowers := []*types.OperatorActivePowerInfo{{}}
+			operatorPowerTotal := sdkmath.LegacyNewDec(0)
+			for _, res := range value {
+				// Find signed operators
+				if res.BlsSignature != nil {
+					signedOperatorList = append(signedOperatorList, res.OperatorAddress)
+					if avsAddr == "" {
+						avsInfo := wrapper.keeper.GetAVSInfoByTaskAddress(ctx, res.TaskContractAddress)
+						avsAddr = avsInfo.AvsAddress
+					}
+					if taskID == 0 {
+						taskID = res.TaskId
+					}
+					if taskAddr == "" {
+						taskAddr = res.TaskContractAddress
+					}
+					power, err := wrapper.keeper.operatorKeeper.GetOperatorOptedUSDValue(ctx, avsAddr, res.OperatorAddress)
+					if err != nil || power.ActiveUSDValue.IsNegative() {
+						ctx.Logger().Error("Failed to update task result statistics", "task result", taskAddr, "error", err)
+						// Handle the error gracefully, continue to the next
+						continue
+					}
+
+					operatorSelfPower := &types.OperatorActivePowerInfo{
+						OperatorAddr:    res.OperatorAddress,
+						SelfActivePower: power.ActiveUSDValue,
+					}
+					operatorPowers = append(operatorPowers, operatorSelfPower)
+					operatorPowerTotal = operatorPowerTotal.Add(power.ActiveUSDValue)
+				}
+			}
+			taskInfo, err := wrapper.keeper.GetTaskInfo(ctx, strconv.FormatUint(taskID, 10), taskAddr)
+			if err != nil {
+				ctx.Logger().Error("Failed to update task result statistics", "task result", taskAddr, "error", err)
+				// Handle the error gracefully, continue to the next
+				continue
```

Comment:
should we skip this error? this would lead to the task not being handled and it seems that it would never be handled in the future

Reply:
Add a future Todo:There should be a retry mechanism or compensation mechanism to handle cases of failure